proxies/dotnet/Controllers/Location.cs

```diff
@@ -0,0 +1,189 @@
+using System.ComponentModel;
+using Newtonsoft.Json.Linq;
+
+namespace dotnet.Controllers;
+
+public class PostBody
+{
+    [Required]
+    public string Command { get; set; } = "";
+
+    [Required]
+    public bool IsAsync { get; set; } = false;
+
+    [Required]
+    public List<JToken> Params { get; set; } = new List<JToken>{};
+}
+
+public class CommandResult
+{
+    [Required]
+    public string EntityType { get; set; } = "";
+
+    [Required]
+    public object Data { get; set; } = new Object{};
+
+    public List<string>? Logs { get; set; } = new List<string>{};
+
+    public string? CommandId { get; set; }
+}
+
+[ApiController]
+[Route("{**catchall}")]
+public class LocationController : ControllerBase
+{
+
+    private readonly ILogger<LocationController> _logger;
+    public LocationController(ILogger<LocationController> logger)
+    {
+        _logger = logger;
+    }
+
+    private static string[] EntityTypes = new[] {"User","Variable","Feature", "Object", "Client"};
+
+    [HttpPost]
+    [Route("/{**location}")]
+    public async Task<object> Post([FromBody] PostBody body, string location)
+    {
+        if (location.Length == 0) {
+            Response.StatusCode = 400;
+            return new { message = "Invalid request: missing location" };
+        }
+
+        object entity;
+        List<object> parsedParams;
+
+        try {
+           entity = GetEntity(location);
+           parsedParams = ParseParams(body.Params);
+        } catch (Exception e) {
+            Response.StatusCode = 404;
+            return new { message = e.Message };
+        }
+
+        try {
+            var result = await InvokeCommand(entity, body.Command, body.IsAsync, parsedParams);
+
+            Response.StatusCode = 201;
+            if (result.CommandId != null) Response.Headers.Add("Location", $"command/{body.Command}/{result.CommandId}");
+
+            return new {
+                entityType = result.EntityType,
+                data = result.Data,
+                logs = result.Logs
+            };
+
+        } catch (Exception e) {
+            Response.StatusCode = 200;
+
+            if (body.IsAsync) {
+                return new {
+                    asyncError = e.Message,
+                    stack = e.StackTrace,
+                };
+            } else {
+                return new {
+                    exception = e.Message,
+                    stack = e.StackTrace,
+                };
+            }
+        }
+    }
+
+    private List<object> ParseParams(List<JToken> bodyParams) {
+        var result = new List<object>{};
+
+        foreach (var param in bodyParams) {
+            if (param["location"] != null) {
+                result.Add(GetEntity(param["location"].ToString()));
+                Console.WriteLine(Newtonsoft.Json.JsonConvert.SerializeObject(GetEntity(param["location"].ToString())));
+            } else if (param["value"] != null) {
+                if (param["value"].Type == JTokenType.Float) {
```

Comment:
is this kinda like a generic JSON cast of the data to a native type?

Reply:
yeah, this is what we spent all afternoon trying to get working yesterday

Reply:
nice